R/utils-json.R

```diff
@@ -169,47 +169,74 @@ getApprovals <- function(data, chain_nm, legend_nm, appr_var_all, month=NULL, po
   approvals_all <- list()
   
   if(approvalsAtBottom==FALSE) {
-    for(approval in appr_type){
-      appr_var <- appr_var_all[which(appr_type == approval)]
-      
-      if(subsetByMonth){
-        points <- subsetByMonth(getTimeSeries(data, chain_nm), month)
-      } else {
-        points <- data[[chain_nm]][['points']]
-        points$time <- formatDates(points[['time']], type=NA)
-      }
-      
-      appr_dates <- getApprovalDates(data, chain_nm, approval)
-      date_index <- apply(appr_dates, 1, function(d, points){
-        which(points$time >= d[1] & points$time <= d[2])}, 
-        points=points)
-      
-      if(is.list(date_index)){
-        date_index_list <- date_index
-      } else {
-        date_index_list <- list(date_index)
-      }
+    approved_dates <- getApprovalDates(data, chain_nm, "Approved")
+    review_dates <-getApprovalDates(data, chain_nm, "In Review")
+    working_dates <- getApprovalDates(data, chain_nm, "Working")
       
+    if(subsetByMonth){
+      points <- subsetByMonth(getTimeSeries(data, chain_nm), month)
+      points_no_times <- points
```

Comment:
178 & 179 can be moved outside of the logic because they are the same as 182 & 183